### ISSUE TITLE
Error: Internal error: Dictionary is full! Execution halted

### ISSUE THREAD

**Issue by gireeshkbogu** [tidyverse/tidyr#713]
Hi, I am trying pivot_wider on a big data file with 1 million rows and 4 columns and it throwing this following error 
```
Error: Internal error: Dictionary is full!
Execution halted
```
I increased the specs on Linux cluster to 1TB memory and 300 TB disk space (working directory) and still the same error. Any solution to this problem would be greatly appreciated. 

FYI: pivot_wider works fine when I run it on 10 % of the input data.

**Comment by hadley** [comment]
Can you please provide a minimal reprex (reproducible example)? The goal of a reprex is to make it as easy as possible for me to recreate your problem so that I can fix it: please [help me help you](https://nypdecider.files.wordpress.com/2014/08/help-me-help-you.gif)! 

If you've never heard of a reprex before, start by reading "[What is a reprex](http://reprex.tidyverse.org/articles/reprex.html)", and follow the advice further down the page. Please make sure your reprex is created with the reprex package as it gives nicely formatted output and avoids a number of common pitfalls.

**Comment by lionel-** [comment]
Maybe caused by lack of support for long vectors in vctrs?

**Comment by gireeshkbogu** [comment]
I used the following code and the data can be downloaded from this link
https://public-docs.crg.es/rguigo/Data/gbogu/Repeat_paper_data/

```
library(tidyverse)
── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 3.0.0           ✔ purrr   0.3.2      
✔ tibble  2.1.3           ✔ dplyr   0.8.3      
✔ tidyr   0.8.99.9000     ✔ stringr 1.3.1      
✔ readr   1.1.1           ✔ forcats 0.3.0      
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
> #devtools::install_github("tidyverse/tidyr")
library(tidyr)

x.a <- readRDS("111222_tes_8051_samples.matrix.main.ann.backup.rds")

x.r <- x.a %>% 
  select(X001aa_SRRID, starts_with("chr"))  

rm4 <- x.r %>%
  pivot_longer(
    cols = starts_with("chr")
  ) %>% select(name, X001aa_SRRID, value)

rm5 <- rm4 %>%
  pivot_wider(names_from = X001aa_SRRID, values_from = value)

```
Error: Internal error: Dictionary is full!
Execution halted

**Comment by gireeshkbogu** [comment]
Maybe this is helpful to you - When I strace, it ends with something like this (shown below). There's no failed mmap() call, so it had no memory issues. Maybe somewhere internal to tidyR something is failing.

```
munmap(0x2aea9bd81000, 8192)            = 0
munmap(0x2af087c99000, 3581796352)      = 0
mmap(NULL, 3581796352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2af087c99000
munmap(0x2aefb24bb000, 3581796352)      = 0
mmap(NULL, 3581796352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2aefb24bb000
mmap(NULL, 7163588608, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2af232c55000
mmap(NULL, 3581796352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2af3ddc10000
munmap(0x2af087c99000, 3581796352)      = 0
mmap(NULL, 7163588608, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2af4b33ee000
getrusage(RUSAGE_SELF, {ru_utime={1072, 857252}, ru_stime={64, 249126}, ...}) = 0
getrusage(RUSAGE_CHILDREN, {ru_utime={0, 9405}, ru_stime={0, 66688}, ...}) = 0
mmap(NULL, 3581796352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2af087c99000
munmap(0x2aeab1589000, 7163588608)      = 0
munmap(0x2aec5c544000, 3581796352)      = 0
munmap(0x2af15d477000, 3581796352)      = 0
munmap(0x2aed31d22000, 3581796352)      = 0
munmap(0x2aee07500000, 7163588608)      = 0
munmap(0x2af3ddc10000, 3581796352)      = 0
write(2, "Error: Internal error: Dictionar"..., 43Error: Internal error: Dictionary is full!
) = 43
write(2, "Execution halted\n", 17Execution halted
)      = 17
```

**Comment by hadley** [comment]
This might already be fixed with dev vctrs.

**Comment by lionel-** [comment]
@DavisVaughan fixed this in r-lib/vctrs#1363.
